sandbox/samples/cpp/alembicassembly/alembicassembly.h

```diff
@@ -0,0 +1,148 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+#ifndef APPLESEED_ALEMBIC_ASSEMBLY_H
+#define APPLESEED_ALEMBIC_ASSEMBLY_H
+
+#define APPLESEED_ENABLE_IMATH_INTEROP
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/modeling/entity/connectableentity.h"
+#include "renderer/utility/transformsequence.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/dual.h"
+#include "foundation/math/vector.h"
+#include "foundation/platform/compiler.h"
+#include "foundation/utility/uid.h"
+
+#include <Alembic/Abc/All.h>
+
+// appleseed.main headers.
+#include "main/dllsymbol.h"
+
+namespace asf = foundation;
+namespace asr = renderer;
+
+namespace
+{
+
+//
+// Alembic Assembly
+//
+
+class AlembicAssembly
+  : public asr::ProceduralAssembly
+{
+  public:
+    // Constructor.
+    AlembicAssembly(
+        const char*                     name,
+        const asr::ParamArray&          params);
+
+    // Return a string identifying the model of this entity.
+    const char* get_model() const override
+    {
+        return Model;
+    };
+
+    // Delete this instance.
+    void release() override
+    {
+        delete this;
+    }
+
+    // This method is called once before rendering each frame.
+    // Returns true on success, false otherwise.
+    /*bool on_frame_begin(
+        const Project&                  project,
+        const BaseGroup*                parent,
+        OnFrameBeginRecorder&           recorder,
+        foundation::IAbortSwitch*       abort_switch = nullptr) override;*/
+
+    // Expand the contents of the assembly.
+    bool do_expand_contents(
+        const asr::Project&     project,
+        const asr::Assembly*    parent,
+        asf::IAbortSwitch*      abort_switch = 0) override;
+
+  private:
+
+    // retrieve and store assembly parameters
+    void retrieve_params(const asr::Project& project);
+
+    void foo(const asr::Assembly* assembly, Alembic::Abc::IObject o);
+
+    static const char* Model;
```

Comment:
You don't need that. Just declare a _single_ `Model` constant in an anonymous namespace in the cpp file, and use it in all `get_model()` methods.